// tailwindcss-addons/dist/index.js
export { flattenColorPalette } from "./lib/flattenColorPalette.js";
export { sizeToRem } from "./lib/sizeToRem.js";
import bgGrid from "./plugins/bgGrid.js";
import bgRadial from "./plugins/bgRadial.js";
import dir from "./plugins/dir.js";
import drag from "./plugins/drag.js";
import extraDefaults from "./plugins/extraDefaults.js";
import flexGrid from "./plugins/flexGrid.js";
import flip from "./plugins/flip.js";
import fontRegular from "./plugins/fontRegular.js";
import hideShow from "./plugins/hideShow.js";
import hocus from "./plugins/hocus.js";
import insetCenter from "./plugins/insetCenter.js";
import not from "./plugins/not.js";
import overflowUnset from "./plugins/overflowUnset.js";
import resetNumberInput from "./plugins/resetNumberInput.js";
import resetSearchInput from "./plugins/resetSearchInput.js";
import scrollbarColor from "./plugins/scrollbarColor.js";
import scrollbarWidth from "./plugins/scrollbarWidth.js";
import supportsFocus from "./plugins/supportsFocus.js";
import supportsHocus from "./plugins/supportsHocus.js";
import supportsHover from "./plugins/supportsHover.js";
import tap from "./plugins/tap.js";
export { bgGrid, bgRadial, dir, drag, extraDefaults, flexGrid, flip, fontRegular, hideShow, hocus, insetCenter, not, overflowUnset, resetNumberInput, resetSearchInput, scrollbarColor, scrollbarWidth, supportsFocus, supportsHocus, supportsHover, tap, };
export const allAddons = (options) => [
    ...((options?.bgGrid ?? true) ? [bgGrid] : []),
    ...((options?.bgRadial ?? true) ? [bgRadial] : []),
    ...((options?.dir ?? true) ? [dir] : []),
    ...((options?.drag ?? true) ? [drag] : []),
    ...((options?.extraDefaults ?? true) ? [extraDefaults] : []),
    ...((options?.flexGrid ?? false) ? [flexGrid] : []),
    ...((options?.flip ?? true) ? [flip] : []),
    ...((options?.fontRegular ?? false) ? [fontRegular] : []),
    ...((options?.hideShow ?? true) ? [hideShow] : []),
    ...((options?.hocus ?? true) ? [hocus] : []),
    ...((options?.insetCenter ?? true) ? [insetCenter] : []),
    ...((options?.not ?? true) ? [not] : []),
    ...((options?.overflowUnset ?? true) ? [overflowUnset] : []),
    ...((options?.resetNumberInput ?? true)
        ? [
            typeof options?.resetNumberInput === "object"
                ? resetNumberInput(options?.resetNumberInput)
                : resetNumberInput(),
        ]
        : []),
    ...((options?.resetSearchInput ?? true)
        ? [
            typeof options?.resetSearchInput === "object"
                ? resetSearchInput(options?.resetSearchInput)
                : resetSearchInput(),
        ]
        : []),
    ...((options?.scrollbarColor ?? true) ? [scrollbarColor] : []),
    ...((options?.scrollbarWidth ?? true) ? [scrollbarWidth] : []),
    ...((options?.supportsFocus ?? true) ? [supportsFocus] : []),
    ...((options?.supportsHocus ?? true) ? [supportsHocus] : []),
    ...((options?.supportsHover ?? true) ? [supportsHover] : []),
    ...((options?.tap ?? true) ? [tap] : []),
];
